test/AWS.Deploy.CLI.UnitTests/ProjectDefinitionParserTest.cs

```diff
@@ -0,0 +1,98 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using AWS.Deploy.CLI.UnitTests.Utilities;
+using AWS.Deploy.Common;
+using AWS.Deploy.Common.IO;
+using Xunit;
+using Should;
+
+namespace AWS.Deploy.CLI.UnitTests
+{
+    public class ProjectDefinitionParserTest
+    {
+        [Theory]
+        [InlineData("WebAppWithDockerFile", "WebAppWithDockerFile.csproj")]
+        [InlineData("WebAppNoDockerFile", "WebAppNoDockerFile.csproj")]
+        [InlineData("ConsoleAppTask", "ConsoleAppTask.csproj")]
+        [InlineData("ConsoleAppService", "ConsoleAppService.csproj")]
+        [InlineData("MessageProcessingApp", "MessageProcessingApp.csproj")]
+        [InlineData("ContosoUniversityBackendService", "ContosoUniversityBackendService.csproj")]
+        [InlineData("ContosoUniversityWeb", "ContosoUniversity.csproj")]
+        [InlineData("BlazorWasm31", "BlazorWasm31.csproj")]
+        [InlineData("BlazorWasm50", "BlazorWasm50.csproj")]
+        public async Task ParseProjectDefinitionWithRelativeProjectPath(string projectName, string csprojName)
+        {
+            //Arrange
+            var currrentWorkingDirectory = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location); 
+            var projectDirectoryPath = SystemIOUtilities.ResolvePath(projectName);
+            var absoluteProjectDirectoryPath = new DirectoryInfo(projectDirectoryPath).FullName;
+            var absoluteProjectPath = Path.Combine(absoluteProjectDirectoryPath, csprojName);
+            var relativeProjectDirectoryPath = Path.GetRelativePath(currrentWorkingDirectory, absoluteProjectDirectoryPath);
+
+            ProjectFileNotFoundException exception = null;
+            ProjectDefinition projectDefinition = null;
+
+            // Act
+            try
+            {
+                projectDefinition = await new ProjectDefinitionParser(new FileManager(), new DirectoryManager())
+                    .Parse(relativeProjectDirectoryPath);
+            }
+            catch (ProjectFileNotFoundException ex)
+            {
+                exception = ex;
+            }
```

Comment:
This is unnecessary, you can remove try-catch. If the method throws, we will know that something is broken.

In test we should only check for throws.

Reply:
Fixed